operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ConfigurationServiceProvider.java

```diff
@@ -7,10 +7,8 @@
  * to the ConfigurationService is via the reconciliation context.
  */
 public class ConfigurationServiceProvider {
-  static final ConfigurationService DEFAULT =
-      new BaseConfigurationService(Utils.loadFromProperties());
   private static ConfigurationService instance;
-  private static ConfigurationService defaultConfigurationService = DEFAULT;
+  private static ConfigurationService defaultConfigurationService = createDefault();
```

Comment:
Yes, this solves an issue with the integration tests, basically since it is singleton, it needs to be reseted after the integration tests. But if the DEFAULT is not created again (with this method in reset) , the configurations are for dependents stay in there and causes trouble. This is just for tests.


see: https://github.com/java-operator-sdk/java-operator-sdk/blob/0b7dc97de05df069e2db87384a9d1f67b3e79724/operator-framework-junit5/src/main/java/io/javaoperatorsdk/operator/junit/AbstractOperatorExtension.java#L165-L165